os/mm/mm_heap/mm_check_heap_corruption.c

```diff
@@ -102,12 +102,12 @@ static void dump_node(struct mm_allocnode_s *node, node_type_t type)
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
 #if CONFIG_TASK_NAME_SIZE > 0
 	if (prctl(PR_GET_NAME, myname, node->pid) == OK) {
-		dbg("Node owner pid = %u (%s), allocated by code at addr = 0x%08x\n", node->pid, myname, node->alloc_call_addr);
+		dbg("Node owner pid = %d (%s), allocated by code at addr = 0x%08x\n", node->pid, myname, node->alloc_call_addr);
 	} else {
-		dbg("Node owner pid = %u (Exited Task), allocated by code at addr = 0x%08x\n", node->pid, node->alloc_call_addr);
+		dbg("Node owner pid = %d (Exited Task), allocated by code at addr = 0x%08x\n", node->pid, node->alloc_call_addr);
```

Comment:
If node->pid is negative, it's not `Exited Task`. How about splitting negative case for `node->pid`?

Reply:
And please consider `prctl` argument for `node->pid` as well for negative.

Reply:
Fixed. Thanks.